sxg_rs/src/wasm_worker.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::headers::AcceptFilter;
+use crate::http::HttpResponse;
+use crate::utils::to_js_error;
+use crate::SxgWorker;
+use anyhow::Result;
+use js_sys::Function as JsFunction;
+use js_sys::Promise as JsPromise;
+use js_sys::Uint8Array;
+use std::sync::Arc;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_futures::future_to_promise;
+
+#[wasm_bindgen]
+pub struct WasmWorker(Arc<SxgWorker>);
+
+#[wasm_bindgen]
+impl WasmWorker {
+    #[wasm_bindgen(constructor)]
+    pub fn new(config_yaml: &str, cert_pem: &str, issuer_pem: &str) -> Result<WasmWorker, JsValue> {
+        let sxg_worker = SxgWorker::new(config_yaml, cert_pem, issuer_pem).map_err(to_js_error)?;
+        Ok(WasmWorker(Arc::new(sxg_worker)))
+    }
+    #[wasm_bindgen(js_name=fetchOcspFromCa)]
+    pub fn fetch_ocsp_from_ca(&self, fetcher: JsFunction) -> JsPromise {
+        let worker = self.0.clone();
+        future_to_promise(async move {
+            let fetcher = crate::fetcher::js_fetcher::JsFetcher::new(fetcher);
+            let response = worker.fetch_ocsp_from_ca(fetcher).await;
+            let ocsp = Uint8Array::from(response.as_slice());
+            // TODO: check whether this conversion has additional cost
```

Comment:
I added this because the old code directly returned `Uint8Array` but the new code returns `JsValue`. Since we do a lot of `JsValue::from`, this particular one does not need special investigating. I have removed the todo.